crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -985,16 +973,34 @@ impl Entities {
         self.meta.len()
     }
 
+    /// The count of all entities in the [`World`] that are used,
+    /// including both those allocated and those reserved, but not those freed.
+    ///
+    /// [`World`]: crate::world::World
+    #[inline]
+    pub fn used_count(&self) -> usize {
+        (self.meta.len() as isize - self.free_cursor.load(Ordering::Relaxed) as isize) as usize
+    }
+
+    /// The count of all entities in the [`World`] that have ever been allocated or reserved, including those that are freed.
+    ///
+    /// [`World`]: crate::world::World
+    #[inline]
+    pub fn total_prospective_count(&self) -> usize {
+        self.total_count() + (-self.free_cursor.load(Ordering::Relaxed)).min(0) as usize
```

Comment:
Very much a nit, but: You use `self.meta.len()` in `used_count` and `self.total_count()` here, and it took me a minute to realize they were the same value.  It might be nice to write it the same way in both places.  

```suggestion
        self.meta.len() + (-self.free_cursor.load(Ordering::Relaxed)).min(0) as usize
```